### PR TITLE
fix(sdk-core): add new function in BaseCoin to get hash-function

### PR DESCRIPTION
**Pull request by dpkjnr**
Ticket: BG-71536

### CODE REVIEW

**Review by mukeshsp — DISMISSED** [review]

File: modules/sdk-core/src/bitgo/baseCoin/baseCoin.ts [hunk 1]
```diff
@@ -462,4 +463,12 @@ export abstract class BaseCoin implements IBaseCoin {
   getInscriptionBuilder(wallet: Wallet): IInscriptionBuilder {
     throw new NotImplementedError('Inscription Builder is not supported for this coin');
   }
+
+  /**
+   * Function to get coin specific hash function used to generate transaction digests.
+   * @returns {@see Hash} hash function if implemented, otherwise undefined
+   */
+  getHashFunction(): Hash | undefined {
+    return undefined;
```
Comment (mmcshinsky-bitgo) [inline modules/sdk-core/src/bitgo/baseCoin/baseCoin.ts:472]:
```suggestion
    throw new NotImplementedError('getHashFunction is not supported for this coin');
```

Comment (zahin-mohammad) [reply modules/sdk-core/src/bitgo/baseCoin/baseCoin.ts:472]: With this suggestion we can also change the return type to be just `Hash`.

File: modules/sdk-core/src/bitgo/wallet/wallet.ts [hunk 2]
```diff
@@ -2817,14 +2817,7 @@ export class Wallet implements IWallet {
...
     try {
-      const hashAlgorithm = this.coinSpecific()?.hashAlgorithm;
-      let hash: Hash;
-      if (hashAlgorithm && hashAlgorithm.startsWith('sha')) {
-        hash = createHash(hashAlgorithm);
-      } else {
-        hash = createKeccakHash('keccak256');
-      }
-
+      const hash: Hash = this.baseCoin.getHashFunction() ?? (createKeccakHash('keccak256') as Hash);
```
Comment (zahin-mohammad) [inline modules/sdk-core/src/bitgo/wallet/wallet.ts:2820]: If we are using `createKeccakHash('keccak256') as Hash` as the default, why not add this to baseCoin?

Comment (dpkjnr) [reply modules/sdk-core/src/bitgo/wallet/wallet.ts]: done

File: modules/sdk-core/src/bitgo/baseCoin/baseCoin.ts (hunk 1 above)
Comment (dpkjnr) [reply modules/sdk-core/src/bitgo/baseCoin/baseCoin.ts]: done

**Review by mukeshsp — DISMISSED** [review]

File: modules/abstract-eth/src/abstractEthLikeCoin.ts [hunk 3]
```diff
@@ -221,4 +222,11 @@ export abstract class AbstractEthLikeCoin extends BaseCoin {
    * @return a new transaction builder
    */
   protected abstract getTransactionBuilder(): TransactionBuilder;
+
+  /**
+   * @inheritdoc
+   */
+  getHashFunction(): Hash {
+    return createKeccakHash('keccak256');
+  }
```
Comment (mullapudipruthvik) [inline modules/abstract-eth/src/abstractEthLikeCoin.ts:231]: are we using this for any coin apart from Atom?
if not any we throw not implemented and get rid of `keccak`

File: modules/abstract-eth/package.json [hunk 4]
```diff
@@ -41,6 +41,8 @@
     "@bitgo/sdk-core": "^7.0.1",
     "@bitgo/statics": "^14.4.0",
     "@bitgo/utxo-lib": "^7.7.0",
-    "bignumber.js": "^8.0.1"
+    "bignumber.js": "^8.0.1",
+    "@types/keccak": "^3.0.1",
```
Comment (mullapudipruthvik) [inline modules/abstract-eth/package.json:45]: `@types/keccak` should be a dev dependency

Comment (dpkjnr) [reply modules/abstract-eth/package.json]: not required after removing default implementation of getHashFunction in abstractEth

**Review by mullapudipruthvik — APPROVED** [review]

**Review by sachushaji — APPROVED** [review]
LGTM

**Review by mukeshsp — APPROVED** [review]

**Pull request merged**